test/Chainweb/Test/Pact/PactInProcApi.hs

```diff
@@ -280,6 +281,240 @@ newBlockRewindValidate mpRefIO reqIO = testCase "newBlockRewindValidate" $ do
             $ mkExec' "(chain-data)"
       }
 
+pact420UpgradeTest :: TestBlockDb -> IO (IORef MemPoolAccess) -> WebPactExecutionService -> IO ()
+pact420UpgradeTest bdb mpRefIO pact = do
+
+  -- run past genesis, upgrades
+  forM_ [(1::Int)..3] $ \_i -> runCut'
+
+  -- run block 7
+  setMempool mpRefIO getBlock4
+  runCut'
+  pwo4 <- getPWO bdb cid
+
+  tx4_0 <- txResult 0 pwo4
+  assertEqual "Hash of coin @ block 4" (pHash "ut_J_ZNkoyaPUEJhiwVeWnkSQn9JT9sQCWKdjjVVrWo") (_crResult tx4_0)
+  assertEqual "Events for tx 0 @ block 4" [] (_crEvents tx4_0)
+
+  tx4_1 <- txResult 1 pwo4
+  assertEqual "Events for tx 1 @ block 4" [] (_crEvents tx4_1)
+
+  tx4_2 <- txResult 2 pwo4
+  assertEqual
+    "Should not resolve new pact natives"
+    (Just "Cannot resolve fold-db")
+    (tx4_2 ^? crResult . to _pactResult . _Left . to peDoc)
+
+  tx4_3 <- txResult 3 pwo4
+  assertEqual
+    "Should not resolve new pact natives"
+    (Just "Cannot resolve zip")
+    (tx4_3 ^? crResult . to _pactResult . _Left . to peDoc)
+
+  tx4_4 <- txResult 4 pwo4
+  assertEqual
+    "Load fdb module"
+    Nothing
+    (tx4_4 ^? crResult . to _pactResult . _Left . to peDoc)
+
+  cb4 <- cbResult pwo4
+  assertEqual "Coinbase events @ block 4" [] (_crEvents cb4)
+
+  -- run past v3 upgrade, pact 4 switch
```

Comment:
Not necessary to run to 20 blocks, that's for pact4 upgrade. See other removals/cleanup and comment in `Version.hs` and make this as low as possible (ie, just running the next block)